packages/client/src/schema/index.ts

```diff
@@ -48,7 +53,7 @@ export class SchemaPlugin<Schemas extends Record<string, XataRecord>> extends Xa
     // Inject generated tables for shell to auto-complete
     const tableNames = pluginOptions.tables?.map(({ name }) => name) ?? [];
     for (const table of tableNames) {
-      db[table] = new RestRepository({ db, pluginOptions, table, schemaTables: pluginOptions.tables });
+      db[table] = new KyselyRepository({ db, pluginOptions, table, schemaTables: pluginOptions.tables });
```

Comment:
To be able to merge this I wonder if we can make it incrementally with a client option that allows us to choose between the old mode and the new one.